docs/wiki/framework/spectator-framework.md

```diff
@@ -0,0 +1,280 @@
+- **Vision modes:** Which vision modes (Normal, Night and Thermal) are available to use by default.
+- **Add Map Locations**: Whether to automatically add the map locations (cities, villages, etc.) to the locations UI list.
+
+Note that these settings govern the default spectator behaviour. Through use of public functions the spectator experience can be more finely controlled (e.g. to limit each player to only be able to spectate units on their side) on top of these.
+
+## 3. Events
+
+### 3.1 Listenable
+
+Event Name | Passed Parameter(s) | Locality | Description
+---------- | ----------- | ------------------- | --------
+`ace_spectatorSet` | [_isSpectator, _player] | Global | Player's spectator status was changed
+
+## 4. Scripting
+
+While the majority of scenarios can be set up as desired using only the settings provided, some require more specific limitations on what functionality is available to whom. All of the following public functions are provided to try and accomidate for such needs.
+
+### 4.1 Change the local player's spectator status
+
+`ace_spectator_fnc_setSpectator`
+If the interface is not forced then the player can close spectator with the <kbd>ESC</kbd> key.
+
+If the player is hidden they will become invisible, invulnerable and removed from their group until they exit spectator.
+
+```cpp
+ * Arguments:
+ * 0: Spectator state of local client <BOOL> (default: true)
+ * 1: Force interface <BOOL> (default: true)
+ * 2: Hide player (if alive) <BOOL> (default: true)
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [true] call ace_spectator_fnc_setSpectator
+```
+
+### 4.2 Whitelist/Blacklist units available to spectate
+
+`ace_spectator_fnc_updateUnits`
+Whitelisted units will always show in the unit list regardless of the usual filtering applied (with the exception of the local player, who will never be shown). Blacklisted units will always be filtered out of the unit list (this overrides the whitelist).
+
+Note that this function takes local effect, so only the local player will see these changes.
+
+```cpp
+ * Arguments:
+ * 0: Units to show in the list <ARRAY>
+ * 1: Units to hide in the list <ARRAY>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [allPlayers, [player]] call ace_spectator_fnc_updateUnits
+ ```
+ 
+### 4.3 Add/Remove sides available to spectate
+
+`ace_spectator_fnc_updateSides`
+By default, units on all 4 sides (`west`, `east`, `independent` and `civilian`) are available to spectate provided they pass through basic filtering (AI setting, simulated and visible). By removing a side, you effectively add a layer to the filtering that removes units belonging to that side from the spectator unit list. Adding sides is simply available to resverse a removal - you cannot add a side beyond the main 4 (e.g. `sideLogic`) as those units do not appear in the `allUnits` array, however you could whitelist a unit belonging to such a side.
```

Comment:
* `effectively add a layer to the filtering` -> `effectively add a layer of filtering`.
* `resverse` -> `reverse`.
* Comma (`,`) after `however`.